libnetwork/iptables/firewalld.go

```diff
@@ -159,13 +182,23 @@ func OnReloaded(callback func()) {
 }
 
 // Call some remote method to see whether the service is actually running.
-func checkRunning() bool {
+func checkRunning(ctx context.Context) (bool, error) {
+	var err error
 	if connection == nil {
-		return false
+		return false, nil
+	}
+
+	var currentState string
+	err = connection.sysObj.CallWithContext(ctx, "org.freedesktop.DBus.Properties.Get", 0, dbusInterface, "state").Store(&currentState)
+
+	if err != nil {
```

Comment:
More idiomatic to do;

```suggestion
	err := connection.sysObj.CallWithContext(ctx, "org.freedesktop.DBus.Properties.Get", 0, dbusInterface, "state").Store(&currentState)
	if err != nil {
```

And removing the `var err error` declaration at the start